python/mleap/sklearn/feature_extraction/text.py

```diff
@@ -65,4 +83,74 @@ def serialize_to_bundle(self, transformer, path, model_name):
                   "port": "output"
                    }]
 
-        self.serialize(transformer, path, model_name, attributes, inputs, outputs)
\ No newline at end of file
+        self.serialize(transformer, path, model_name, attributes, inputs, outputs)
+
+
+class TfidfVectorSerializer(MLeapSerializer):
+    pipeline_serializer = PipelineSerializer()
+
+    def __init__(self):
+        super(TfidfVectorSerializer, self).__init__()
+
+    @staticmethod
+    def set_prediction_column(transformer, prediction_column):
+        transformer.prediction_column = prediction_column
+
+    @staticmethod
+    def set_input_features(transformer, input_features):
+        transformer.input_features = input_features
+
+    def serialize_to_bundle(self, transformer, path, model_name):
+        num_features = transformer.idf_.shape[0]
+        vocabulary = [None] * num_features
+
+        for term, termidx in transformer.vocabulary_.iteritems():
+            vocabulary[termidx] = str(term)
+
+        tf_attributes = [
+            ('vocabulary', vocabulary),
+            ('binary', transformer.binary),
+            # no concept of 'min_tf' in scikit's TfidfVectorizer
+            ('min_tf', None)
+        ]
+        tf_inputs = [{
+            "name": transformer.input_features,
+            "port": "input"
+        }]
+        tf_outputs = [{
+            "name": "{}_count_vectorizer".format(transformer.name),
```

Comment:
I think we can just put the output name of the count vectorizer to be something like 'token_counts' or something similar... it'll make it a bit easier to read if it doesn't have the transformer name in it

Reply:
Sounds good!